test-clients/src/main/java/com/hedera/services/bdd/spec/queries/HapiQueryOp.java

```diff
@@ -93,14 +95,17 @@ protected long costFrom(Response response) throws Throwable {
 			if (permissibleCostAnswerPrechecks.get().contains(actualPrecheck)) {
 				costAnswerPrecheck = Optional.of(actualPrecheck);
 			} else {
-				Assert.fail(
-						String.format(
-								"Cost-answer precheck was %s, not one of %s!",
-								actualPrecheck,
-								permissibleCostAnswerPrechecks.get()));
+				String errMsg = String.format("Cost-answer precheck was %s, not one of %s!",
+						actualPrecheck,	permissibleCostAnswerPrechecks.get());
+				log.error(errMsg);
+
+				throw new HapiQueryCheckStateException(errMsg);
 			}
 		} else {
-			Assert.assertEquals("Bad costAnswerPrecheck!", expectedCostAnswerPrecheck(), actualPrecheck);
+			if(expectedCostAnswerPrecheck() != actualPrecheck) {
+				log.error("Bad costAnswerPrecheck! expected {}, actual {}", expectedCostAnswerPrecheck(), actualPrecheck);
+				throw new HapiQueryCheckStateException(String.format("Bad costAnswerPrecheck! expected %s, actial %s", expectedAnswerOnlyPrecheck(), actualPrecheck));
```

Comment:
typo `actial` --> `actual`
Should use String.format as in other places to remove duplicate code

Reply:
Done